pulp_rpm/tests/functional/api/test_sync.py

```diff
@@ -487,6 +490,115 @@ def test_file_decriptors(self):
         response = cli_client.run(cmd).stdout
         self.assertEqual(len(response), 0, response)
 
+    def test_optimize(self):
+        """Sync is no-op without changes.
+
+        This test targets the following issue:
+
+        `Pulp #6313 <https://pulp.plan.io/issues/6313>`_
+
+        If there are no changes, a full sync should not be done.
+
+        Do the following:
+
+        1. Sync (a repo and a remote will be created automatically).
+        2. Sync again to get our task dictionary.
+        3. Assert an "Optimizing Sync" progress report is present.
+        4. Sync again with flag "optimize=False".
+        5. Assert "Optimizing Sync" progress report is absent.
+        # 6. **Create a new repo version.
+        # 7. Sync again (no flag).
+        # 8. Assert "Optimizing Sync" progress report is absent.
+        9. Update remote to have "policy=immediate".
+        10. Sync again.
+        11. Assert "Optimizing Sync" progress report is absent.
+        12. Create a new (different) remote.
+        13. Sync using the new remote.
+        14. Assert "Optimizing Sync" progress report is absent.
+        # 15. Sync again with the new remote.
+        # 16. Assert an "Optimizing Sync" progress report is present.
+        """
+        # sync
+        repo, remote = self.do_test()
+
+        # add resources to clean up
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # sync, get progress reports from task
+        report_list = self.sync(repository=repo, remote=remote)
+
+        # check for an optimization progress report
+        optimized = self.optimize_report(progress_reports=report_list)
+
+        # check that sync was optimized
+        self.assertTrue(optimized)
+
+        # sync again with flag optimize=False
+        report_list = self.sync(repository=repo, remote=remote, optimize=False)
+
+        # check for an optimization progress report
+        optimized = self.optimize_report(progress_reports=report_list)
+
+        # check that sync was not optimized
+        self.assertFalse(optimized)
+
+        # # create a new repo version
```

Comment:
As discussed on our call, I recommend just removing a content unit to create a new repository version. Here is an example from pulpcore: https://github.com/pulp/pulpcore/blob/master/pulpcore/tests/functional/api/using_plugin/test_repo_versions.py#L160